src/main/java/meetup/connect/googleApi/GoogleCalendar.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+
+@Service
+public class GoogleCalendar {
+  private static final String APPLICATION_NAME = "MeetUpConnect Calendar";
+
+  /** Global instance of the JSON factory. */
+  private static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+
+  /** Directory to store authorization tokens for this application. */
+  private static final String TOKENS_DIRECTORY_PATH = "token";
+  private static final String CIRCUIT_BREAKER_NAME = "googleApiService";
+  private static final Integer PORT = 8888;
+
+  private static final String CALENDAR_ID = "primary";
+  private static final Logger log = LoggerFactory.getLogger(GoogleCalendar.class);
+
+  /**
+   * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+   * your previously saved tokens/ folder.
+   */
+  private static final List<String> SCOPES = List.of(CalendarScopes.CALENDAR_EVENTS);
+
+  private final CalendarProperties calendarProperties;
+  private static Calendar service;
+
+  public GoogleCalendar(CalendarProperties calendarProperties) {
+    this.calendarProperties = calendarProperties;
+  }
+
+  private Credential getCredentials() throws IOException {
+    GoogleAuthorizationCodeFlow flow =
+        new GoogleAuthorizationCodeFlow.Builder(
+                new NetHttpTransport(),
+                GsonFactory.getDefaultInstance(),
+                calendarProperties.getClient_id(),
+                calendarProperties.getClient_secret(),
+                SCOPES)
+            .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+            .setAccessType("offline")
+            .build();
+    LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(PORT).build();
+    // returns an authorized Credential object.
+    return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+  }
+
+  private Calendar getService() throws GeneralSecurityException, IOException {
+    final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    var requestFactory = HTTP_TRANSPORT
+            .createRequestFactory(httpRequest -> httpRequest
+                    .setConnectTimeout(10)
+                    .setReadTimeout(5)
+                    .setNumberOfRetries(3)
+            )
+            .getTransport();
+    return new Calendar.Builder(requestFactory, JSON_FACTORY, getCredentials())
+            .setApplicationName(APPLICATION_NAME)
+            .build();
+  }
+
+  private Supplier<Calendar> getGoogleCalendarSupplier() {
+    CircuitBreakerConfig googleApiServiceConfig = CircuitBreakerConfig.ofDefaults();
+    CircuitBreakerRegistry circuitBreakerRegistry = CircuitBreakerRegistry.of(
+            Map.of(CIRCUIT_BREAKER_NAME, googleApiServiceConfig)
+    );
+    CircuitBreaker circuitBreaker = circuitBreakerRegistry.circuitBreaker(CIRCUIT_BREAKER_NAME);
```

Comment:
nit: It would be better to create a bean named `CIRCUIT_BREAKER_NAME` and inject it with `@Qualifier(CIRCUIT_BREAKER_NAME)`

Reply:
https://docs.spring.io/spring-framework/reference/core/beans/java/bean-annotation.html#beans-java-customizing-bean-naming